sdk/messaging/azeventhubs/internal/namespace.go

```diff
@@ -259,9 +261,10 @@ func (ns *Namespace) Recover(ctx context.Context, theirConnID uint64) (bool, err
 	}
 
 	if ns.connID != theirConnID {
-		log.Writef(exported.EventConn, "Skipping connection recovery, already recovered: %d vs %d", ns.connID, theirConnID)
 		// we've already recovered since the client last tried.
-		return false, nil
+		// they should recover their links since their connection is old.
+		log.Writef(exported.EventConn, "Skipping connection recovery, already recovered: %d vs %d. Links will still be recovered.", ns.connID, theirConnID)
+		return true, nil
```

Comment:
The fact that this used to return `false` was just a bug?

Reply:
Yes. I'll be making this same change in azservicebus before we ship the next hotfix.

Reply:
Probably worth noting in the changelog then.

Reply:
Added.

Reply:
Thinking about this more, I want to look at the code. I think it went through some evolutions, and I want to look at it fresh.

Reply:
Talked with @jhendrixMSFT offline - I'm going to revert this because I think there are some more cases I want to cover and test. Will do that in the next PR so we can keep this one simple.